python/pyiceberg/utils/schema_conversion.py

```diff
@@ -468,3 +476,129 @@ def _convert_fixed_type(self, avro_type: Dict[str, Any]) -> FixedType:
+        }
+
+        if field.optional:
+            result["default"] = None
+
+        if field.doc is not None:
+            result["doc"] = field.doc
+
+        return result
+
+    def list(self, list_type: ListType, element_result: AvroType) -> AvroType:
+        # Sets the schema name in case of a record
+        if isinstance(element_result, dict) and element_result.get("type") == "record":
+            element_result["name"] = f"r{self.last_list_field_id}"
+        return {"type": "array", "element-id": self.last_list_field_id, "items": element_result}
+
+    def map(self, map_type: MapType, key_result: AvroType, value_result: AvroType) -> AvroType:
+        if isinstance(key_result, StringType):
+            # Avro Maps does not support other keys than a String,
+            return {
+                "type": "map",
+                "values": value_result,
+            }
+        else:
+            # Creates a logical map that's a list of schema's
+            # binary compatible
+            return {
+                "type": "array",
+                "items": {
+                    "type": "record",
+                    "name": f"k{self.last_map_key_field_id}_v{self.last_map_value_field_id}",
+                    "fields": [
+                        {"name": "key", "type": key_result, "field-id": self.last_map_key_field_id},
+                        {"name": "value", "type": value_result, "field-id": self.last_map_value_field_id},
+                    ],
+                },
+                "logicalType": "map",
+            }
+
+    def visit_fixed(self, fixed_type: FixedType) -> AvroType:
+        return {"type": "fixed", "size": len(fixed_type)}
+
+    def visit_decimal(self, decimal_type: DecimalType) -> AvroType:
+        return {"type": "bytes", "logicalType": "decimal", "precision": decimal_type.precision, "scale": decimal_type.scale}
+
+    def visit_boolean(self, boolean_type: BooleanType) -> AvroType:
+        return "boolean"
+
+    def visit_integer(self, integer_type: IntegerType) -> AvroType:
+        return "int"
+
+    def visit_long(self, long_type: LongType) -> AvroType:
+        return "long"
+
+    def visit_float(self, float_type: FloatType) -> AvroType:
+        return "float"
+
+    def visit_double(self, double_type: DoubleType) -> AvroType:
+        return "double"
+
+    def visit_date(self, date_type: DateType) -> AvroType:
+        return {"type": "int", "logicalType": "date"}
+
+    def visit_time(self, time_type: TimeType) -> AvroType:
+        return {"type": "long", "logicalType": "time-micros"}
+
+    def visit_timestamp(self, timestamp_type: TimestampType) -> AvroType:
+        # Iceberg only supports micro's
+        return {"type": "long", "logicalType": "timestamp-micros"}
+
+    def visit_timestamptz(self, timestamptz_type: TimestamptzType) -> AvroType:
+        # Iceberg only supports micro's
+        return {"type": "long", "logicalType": "timestamp-micros"}
```

Comment:
This needs to set `adjust-to-utc` to `true` to signal that the value is a timestamptz.

Reply:
Great catch, this actually uncovered another bug. Thanks!